### ISSUE TITLE
Error: Looks like "https://kubernetes-charts.storage.googleapis.com" is not a valid chart repository or cannot be reached

### ISSUE THREAD

**Issue by diclophis** [kubernetes/helm#4251]
Hello, I am running into some issues with TLS attempting to add the default stable repo. I am on OSX `10.13.5 (17F77)` and below is the summary of my issue:

```
$ rm -Rf ~/.helm
$ /usr/local/bin/helm init --client-only
Creating /Users/XXX/.helm 
Creating /Users/XXX/.helm/repository 
Creating /Users/XXX/.helm/repository/cache 
Creating /Users/XXX/.helm/repository/local 
Creating /Users/XXX/.helm/plugins 
Creating /Users/XXX/.helm/starters 
Creating /Users/XXX/.helm/cache/archive 
Creating /Users/XXX/.helm/repository/repositories.yaml 
Adding stable repo with URL: https://kubernetes-charts.storage.googleapis.com 
Error: Looks like "https://kubernetes-charts.storage.googleapis.com" is not a valid chart repository or cannot be reached: Get https://kubernetes-charts.storage.googleapis.com/index.yaml: net/http: TLS handshake timeout
```

It appears as though the TLS handshake to the `kubernetes-charts.storage.googleapis.com` host is failing...

Output of `helm version`:

```
$ helm version --client
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}

$ shasum -a 256 /usr/local/bin/helm
bee3146de8b0cf940e74c9b1e7d3e23a1ffd8f7c4b8aad59c3e5a5152d7b818b  /usr/local/bin/helm
```

Output of `kubectl version`:

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.0", GitCommit:"925c127ec6b946659ad0fd596fa959be43f0cc05", GitTreeState:"clean", BuildDate:"2017-12-15T21:07:38Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-21T09:05:37Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 

```
Docker4Mac
Version 18.05.0-ce-mac67 (25042)
Channel: edge
0c7f1aa2fa
Kubernetes: v1.10.3
```

**Comment by SlickNik** [comment]
@diclophis Thanks for reporting this! This seems strange. I'm also on OSX and just checked, but am not having any problems with the TLS handshake to kubernetes-charts.storage.googleapis.com.
```openssl s_client -connect kubernetes-charts.storage.googleapis.com:443```
seems to return the handshake and the certificate.

Can you verify that your network connection is fine, and that you're not behind a proxy of some sort that may be causing issues with the TLS session?

**Comment by diclophis** [comment]
I am able to connect to the repo URL fine in `curl` `chrome` `safari` ... below is the output from the mentioned `openssl` command:

```
$  openssl s_client -connect kubernetes-charts.storage.googleapis.com:443
CONNECTED(00000005)
depth=1 C = US, O = Google Trust Services, CN = Google Internet Authority G3
verify error:num=20:unable to get local issuer certificate
verify return:0
---
Certificate chain
 0 s:/C=US/ST=California/L=Mountain View/O=Google LLC/CN=*.storage.googleapis.com
   i:/C=US/O=Google Trust Services/CN=Google Internet Authority G3
 1 s:/C=US/O=Google Trust Services/CN=Google Internet Authority G3
   i:/OU=GlobalSign Root CA - R2/O=GlobalSign/CN=GlobalSign
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIGajCCBVKgAwIBAgIIUHwZDy112f4wDQYJKoZIhvcNAQELBQAwVDELMAkGA1UE
BhMCVVMxHjAcBgNVBAoTFUdvb2dsZSBUcnVzdCBTZXJ2aWNlczElMCMGA1UEAxMc
R29vZ2xlIEludGVybmV0IEF1dGhvcml0eSBHMzAeFw0xODA2MDcxMDQ4MTNaFw0x
ODA4MTYwOTUwMDBaMHIxCzAJBgNVBAYTAlVTMRMwEQYDVQQIDApDYWxpZm9ybmlh
MRYwFAYDVQQHDA1Nb3VudGFpbiBWaWV3MRMwEQYDVQQKDApHb29nbGUgTExDMSEw
HwYDVQQDDBgqLnN0b3JhZ2UuZ29vZ2xlYXBpcy5jb20wggEiMA0GCSqGSIb3DQEB
AQUAA4IBDwAwggEKAoIBAQC3rB/0dhsN4doSHrGOwLMV1m3TutvamKHV1U3h0fFi
mTaOVEtcH1VzEl9T1DbAXC2WTk8yuYKfnIAbESnTTyIQWjPC5SEhJSgj6T8O7vyR
rCVC/eOhRtruJIMwL8tAntDKufSgHbFlZqNv5mm0VdW0Nvo93Hx1mQOAUzYs5Te+
sZl9pr7fNW31DCV6OQa87LtEMRx1jQWAOUJFoefDr+6RhGVnynKIyFJqPqBB4v1K
QDGC4eJkCMcZOVIXPR7ckgXWQDx4PGrAwTfxG3qOwiQfXkgUnbgvioyx4EtWhDeV
mrCuWAtFnYjte1358K/JwjnzcMVtHTCn4xHegM6UiSurAgMBAAGjggMgMIIDHDAT
BgNVHSUEDDAKBggrBgEFBQcDATCCAfUGA1UdEQSCAewwggHoghgqLnN0b3JhZ2Uu
Z29vZ2xlYXBpcy5jb22CJCouYXBwc3BvdC5jb20uc3RvcmFnZS5nb29nbGVhcGlz
LmNvbYIiKi5jb21tb25kYXRhc3RvcmFnZS5nb29nbGVhcGlzLmNvbYIpKi5jb250
ZW50LXN0b3JhZ2UtZG93bmxvYWQuZ29vZ2xlYXBpcy5jb22CJyouY29udGVudC1z
dG9yYWdlLXVwbG9hZC5nb29nbGVhcGlzLmNvbYIgKi5jb250ZW50LXN0b3JhZ2Uu
Z29vZ2xlYXBpcy5jb22CECouZ29vZ2xlYXBpcy5jb22CISouc3RvcmFnZS1kb3du
bG9hZC5nb29nbGVhcGlzLmNvbYIfKi5zdG9yYWdlLXVwbG9hZC5nb29nbGVhcGlz
LmNvbYIfKi5zdG9yYWdlLnNlbGVjdC5nb29nbGVhcGlzLmNvbYIgY29tbW9uZGF0
YXN0b3JhZ2UuZ29vZ2xlYXBpcy5jb22CK3N0YXRpYy5wYW5vcmFtaW8uY29tLnN0
b3JhZ2UuZ29vZ2xlYXBpcy5jb22CFnN0b3JhZ2UuZ29vZ2xlYXBpcy5jb22CHXN0
b3JhZ2Uuc2VsZWN0Lmdvb2dsZWFwaXMuY29tgg91bmZpbHRlcmVkLm5ld3MwaAYI
KwYBBQUHAQEEXDBaMC0GCCsGAQUFBzAChiFodHRwOi8vcGtpLmdvb2cvZ3NyMi9H
VFNHSUFHMy5jcnQwKQYIKwYBBQUHMAGGHWh0dHA6Ly9vY3NwLnBraS5nb29nL0dU
U0dJQUczMB0GA1UdDgQWBBSq/mLTqw1lC9gufwj945YaJnb9gDAMBgNVHRMBAf8E
AjAAMB8GA1UdIwQYMBaAFHfCuFCaZ3Z2sS3ChtCDoH6mfrpLMCEGA1UdIAQaMBgw
DAYKKwYBBAHWeQIFAzAIBgZngQwBAgIwMQYDVR0fBCowKDAmoCSgIoYgaHR0cDov
L2NybC5wa2kuZ29vZy9HVFNHSUFHMy5jcmwwDQYJKoZIhvcNAQELBQADggEBAKw7
s9liHClp60Bjnt26N8p7jtpA6mrEavaLR4iw1V521H3TyuP2OL6XYXGaW8IUQw9j
Foq/wKuw6IO0trjQBMo2ma7HidLTLeE92NjvzMQ3GvtqRhwWlAKAjdTV3dhHhw80
vy9LVV5oeVoqRGoDmqvnBrhBfluyVoaqEwqsuCi7LMh6tyGUsf4fJkuXUzaHBEB3
KOaIEd+FC70qZhZgDu+8Qios7IdUi1FnHrelUxExMNIQhO2MX4DkuIM5yYo5uPf7
0uVHIVbRmeD93peEknWlbHY7CNg2r5p9z+ySHtJvF/CcmT3P+ptUFg+jODHJj/pJ
2Cy+re9WgRY4Y1JWYfw=
-----END CERTIFICATE-----
subject=/C=US/ST=California/L=Mountain View/O=Google LLC/CN=*.storage.googleapis.com
issuer=/C=US/O=Google Trust Services/CN=Google Internet Authority G3
---
No client certificate CA names sent
---
SSL handshake has read 3474 bytes and written 444 bytes
---
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-AES128-GCM-SHA256
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-AES128-GCM-SHA256
    Session-ID: D0CD0CB8E220990A50337F17D55B9344E1348BC1308CEC31E73F055A43A2EFCC
    Session-ID-ctx: 
    Master-Key: 126C88F0FFED78FC4DDE5E1DC3CD452DCB762D7A959BB4C9F8E044015501C5FBF2D00F2DBE0F192DD44D0F4C94A95E02
    TLS session ticket lifetime hint: 100800 (seconds)
    TLS session ticket:
    0000 - 00 73 12 76 de 2b cd e3-42 e2 ab 0f 29 51 52 f9   .s.v.+..B...)QR.
    0010 - 5d 5e 21 1c 43 7f dc 87-ce 59 3e 7b 31 bc ea e9   ]^!.C....Y>{1...
    0020 - ea 38 6c 80 08 fc a4 2d-d4 8d 8f b3 63 87 eb 12   .8l....-....c...
    0030 - 8f 99 c2 4a 7a 2f e4 00-63 ec b3 bd 87 22 bd 22   ...Jz/..c...."."
    0040 - bc 8a 73 7e a8 bf 55 9e-47 6d 8a a0 44 de 55 63   ..s~..U.Gm..D.Uc
    0050 - b0 da 52 87 36 18 68 59-93 e2 5b a5 d5 ea 3d 80   ..R.6.hY..[...=.
    0060 - a4 7a 19 a1 29 90 1b cb-cf b1 c8 e9 2b 5f 32 a1   .z..).......+_2.
    0070 - ad ac 78 56 7c e7 7a 3b-80 3e 04 4b 56 50 ae f7   ..xV|.z;.>.KVP..
    0080 - 1f d9 2b da 66 e6 e7 83-53 56 e2 4e 6c 8a 78 58   ..+.f...SV.Nl.xX
    0090 - 84 4a a2 83 3e 3f 16 88-0d 34 66 63 82 d2 7d 6e   .J..>?...4fc..}n
    00a0 - 76 f7 e1 d6 71 54 0e 89-1b e5 06 18 79 c8 1c e1   v...qT......y...
    00b0 - 63 ef 98 5a 4f c6 2e f4-a3 95 c7 a3 55 89 e0 c8   c..ZO.......U...
    00c0 - cc a7 90 aa 6b 6b df 19-24 c4 ed 58 da 35 cf 7a   ....kk..$..X.5.z
    00d0 - e9 9d e6 26 57                                    ...&W

    Start Time: 1529580523
    Timeout   : 300 (sec)
    Verify return code: 0 (ok)
---
```

I think it may have something to do with what CA bundle the `helm` binary is loading, though I am not sure how to debug the process further to figure out why it cannot negotiated a TLS connection.

**Comment by bacongobbler** [comment]
I just noticed this line in your output. Maybe this is relevant?

> verify error:num=20:unable to get local issuer certificate

I'm just spiralling here and have not had coffee yet, apologies if this is just a wild goose chase. :Coffee:

**Comment by bacongobbler** [comment]
hmm nope. I see the same thing on MacOS, however I can't reproduce the error.

```
><> rm -rf ~/.helm/
><> helm version -c
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
><> helm init -c
Creating /Users/bacongobbler/.helm
Creating /Users/bacongobbler/.helm/repository
Creating /Users/bacongobbler/.helm/repository/cache
Creating /Users/bacongobbler/.helm/repository/local
Creating /Users/bacongobbler/.helm/plugins
Creating /Users/bacongobbler/.helm/starters
Creating /Users/bacongobbler/.helm/cache/archive
Creating /Users/bacongobbler/.helm/repository/repositories.yaml
Adding stable repo with URL: https://kubernetes-charts.storage.googleapis.com
Adding local repo with URL: http://127.0.0.1:8879/charts
$HELM_HOME has been configured at /Users/bacongobbler/.helm.
Not installing Tiller due to 'client-only' flag having been set
Happy Helming!
```

**Comment by diclophis** [comment]
Maybe this isn't directly related to `helm`... any help figuring this out would be appreciated though... please observe the following go program, two different compilation modes ("normal static" and full static") and how the "full-static" binary fails to connect

### go build -o normal-static main.go

```
$ time DYLD_PRINT_LIBRARIES=YES ./normal-static 
dyld: loaded: /Users/XXX/workspace/go/src/github.com/XXX/XXX/./normal-static
dyld: loaded: /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
dyld: loaded: /System/Library/Frameworks/Security.framework/Versions/A/Security
dyld: loaded: /usr/lib/libSystem.B.dylib
dyld: loaded: /usr/lib/libDiagnosticMessagesClient.dylib
dyld: loaded: /usr/lib/libicucore.A.dylib
dyld: loaded: /usr/lib/libobjc.A.dylib
dyld: loaded: /usr/lib/libz.1.dylib
dyld: loaded: /usr/lib/system/libcache.dylib
dyld: loaded: /usr/lib/system/libcommonCrypto.dylib
dyld: loaded: /usr/lib/system/libcompiler_rt.dylib
dyld: loaded: /usr/lib/system/libcopyfile.dylib
dyld: loaded: /usr/lib/system/libcorecrypto.dylib
dyld: loaded: /usr/lib/system/libdispatch.dylib
dyld: loaded: /usr/lib/system/libdyld.dylib
dyld: loaded: /usr/lib/system/libkeymgr.dylib
dyld: loaded: /usr/lib/system/liblaunch.dylib
dyld: loaded: /usr/lib/system/libmacho.dylib
dyld: loaded: /usr/lib/system/libquarantine.dylib
dyld: loaded: /usr/lib/system/libremovefile.dylib
dyld: loaded: /usr/lib/system/libsystem_asl.dylib
dyld: loaded: /usr/lib/system/libsystem_blocks.dylib
dyld: loaded: /usr/lib/system/libsystem_c.dylib
dyld: loaded: /usr/lib/system/libsystem_configuration.dylib
dyld: loaded: /usr/lib/system/libsystem_coreservices.dylib
dyld: loaded: /usr/lib/system/libsystem_darwin.dylib
dyld: loaded: /usr/lib/system/libsystem_dnssd.dylib
dyld: loaded: /usr/lib/system/libsystem_info.dylib
dyld: loaded: /usr/lib/system/libsystem_m.dylib
dyld: loaded: /usr/lib/system/libsystem_malloc.dylib
dyld: loaded: /usr/lib/system/libsystem_network.dylib
dyld: loaded: /usr/lib/system/libsystem_networkextension.dylib
dyld: loaded: /usr/lib/system/libsystem_notify.dylib
dyld: loaded: /usr/lib/system/libsystem_sandbox.dylib
dyld: loaded: /usr/lib/system/libsystem_secinit.dylib
dyld: loaded: /usr/lib/system/libsystem_kernel.dylib
dyld: loaded: /usr/lib/system/libsystem_platform.dylib
dyld: loaded: /usr/lib/system/libsystem_pthread.dylib
dyld: loaded: /usr/lib/system/libsystem_symptoms.dylib
dyld: loaded: /usr/lib/system/libsystem_trace.dylib
dyld: loaded: /usr/lib/system/libunwind.dylib
dyld: loaded: /usr/lib/system/libxpc.dylib
dyld: loaded: /usr/lib/closure/libclosured.dylib
dyld: loaded: /usr/lib/libc++abi.dylib
dyld: loaded: /usr/lib/libc++.1.dylib
dyld: loaded: /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation
dyld: loaded: /usr/lib/libOpenScriptingUtil.dylib
dyld: loaded: /usr/lib/libauto.dylib
dyld: loaded: /usr/lib/libbsm.0.dylib
dyld: loaded: /usr/lib/libcoretls.dylib
dyld: loaded: /usr/lib/libcoretls_cfhelpers.dylib
dyld: loaded: /usr/lib/libpam.2.dylib
dyld: loaded: /usr/lib/libsqlite3.dylib
dyld: loaded: /usr/lib/libxar.1.dylib
dyld: loaded: /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
dyld: loaded: /System/Library/Frameworks/DiskArbitration.framework/Versions/A/DiskArbitration
dyld: loaded: /usr/lib/libarchive.2.dylib
dyld: loaded: /usr/lib/libxml2.2.dylib
dyld: loaded: /System/Library/Frameworks/CFNetwork.framework/Versions/A/CFNetwork
dyld: loaded: /System/Library/Frameworks/SystemConfiguration.framework/Versions/A/SystemConfiguration
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices
dyld: loaded: /usr/lib/liblangid.dylib
dyld: loaded: /usr/lib/libCRFSuite.dylib
dyld: loaded: /usr/lib/libenergytrace.dylib
dyld: loaded: /usr/lib/system/libkxld.dylib
dyld: loaded: /usr/lib/libbz2.1.0.dylib
dyld: loaded: /usr/lib/liblzma.5.dylib
dyld: loaded: /usr/lib/libnetwork.dylib
dyld: loaded: /usr/lib/libapple_nghttp2.dylib
dyld: loaded: /usr/lib/libpcap.A.dylib
dyld: loaded: /usr/lib/libboringssl.dylib
dyld: loaded: /usr/lib/libusrtcp.dylib
dyld: loaded: /usr/lib/libapple_crypto.dylib
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/FSEvents.framework/Versions/A/FSEvents
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/CarbonCore.framework/Versions/A/CarbonCore
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/Metadata.framework/Versions/A/Metadata
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/OSServices.framework/Versions/A/OSServices
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/SearchKit.framework/Versions/A/SearchKit
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/AE.framework/Versions/A/AE
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/LaunchServices
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/DictionaryServices.framework/Versions/A/DictionaryServices
dyld: loaded: /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/SharedFileList.framework/Versions/A/SharedFileList
dyld: loaded: /System/Library/Frameworks/NetFS.framework/Versions/A/NetFS
dyld: loaded: /System/Library/PrivateFrameworks/NetAuth.framework/Versions/A/NetAuth
dyld: loaded: /System/Library/PrivateFrameworks/login.framework/Versions/A/Frameworks/loginsupport.framework/Versions/A/loginsupport
dyld: loaded: /System/Library/PrivateFrameworks/TCC.framework/Versions/A/TCC
dyld: loaded: /usr/lib/libmecabra.dylib
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/ApplicationServices
dyld: loaded: /System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics
dyld: loaded: /System/Library/Frameworks/CoreText.framework/Versions/A/CoreText
dyld: loaded: /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
dyld: loaded: /System/Library/Frameworks/ColorSync.framework/Versions/A/ColorSync
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ATS.framework/Versions/A/ATS
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ColorSyncLegacy.framework/Versions/A/ColorSyncLegacy
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/HIServices.framework/Versions/A/HIServices
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/LangAnalysis.framework/Versions/A/LangAnalysis
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/PrintCore.framework/Versions/A/PrintCore
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/QD.framework/Versions/A/QD
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/SpeechSynthesis.framework/Versions/A/SpeechSynthesis
dyld: loaded: /System/Library/PrivateFrameworks/SkyLight.framework/Versions/A/SkyLight
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Accelerate
dyld: loaded: /System/Library/Frameworks/CoreDisplay.framework/Versions/A/CoreDisplay
dyld: loaded: /System/Library/Frameworks/IOSurface.framework/Versions/A/IOSurface
dyld: loaded: /System/Library/Frameworks/Metal.framework/Versions/A/Metal
dyld: loaded: /System/Library/PrivateFrameworks/MultitouchSupport.framework/Versions/A/MultitouchSupport
dyld: loaded: /System/Library/Frameworks/QuartzCore.framework/Versions/A/QuartzCore
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vImage.framework/Versions/A/vImage
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/vecLib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libvDSP.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBNNS.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libQuadrature.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libvMisc.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libLAPACK.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libLinearAlgebra.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libSparse.dylib
dyld: loaded: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libSparseBLAS.dylib
dyld: loaded: /System/Library/PrivateFrameworks/GPUWrangler.framework/Versions/A/GPUWrangler
dyld: loaded: /System/Library/PrivateFrameworks/IOAccelerator.framework/Versions/A/IOAccelerator
dyld: loaded: /System/Library/PrivateFrameworks/IOPresentment.framework/Versions/A/IOPresentment
dyld: loaded: /System/Library/PrivateFrameworks/DSExternalDisplay.framework/Versions/A/DSExternalDisplay
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libCoreFSCache.dylib
dyld: loaded: /System/Library/Frameworks/CoreImage.framework/Versions/A/CoreImage
dyld: loaded: /System/Library/Frameworks/CoreVideo.framework/Versions/A/CoreVideo
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/OpenGL
dyld: loaded: /System/Library/PrivateFrameworks/GraphVisualizer.framework/Versions/A/GraphVisualizer
dyld: loaded: /System/Library/Frameworks/MetalPerformanceShaders.framework/Versions/A/MetalPerformanceShaders
dyld: loaded: /usr/lib/libFosl_dynamic.dylib
dyld: loaded: /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/FaceCore
dyld: loaded: /System/Library/Frameworks/OpenCL.framework/Versions/A/OpenCL
dyld: loaded: /usr/lib/libcompression.dylib
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ATS.framework/Versions/A/Resources/libFontParser.dylib
dyld: loaded: /System/Library/Frameworks/ApplicationServices.framework/Versions/A/Frameworks/ATS.framework/Versions/A/Resources/libFontRegistry.dylib
dyld: loaded: /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libJPEG.dylib
dyld: loaded: /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libTIFF.dylib
dyld: loaded: /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libPng.dylib
dyld: loaded: /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libGIF.dylib
dyld: loaded: /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libJP2.dylib
dyld: loaded: /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libRadiance.dylib
dyld: loaded: /System/Library/PrivateFrameworks/AppleJPEG.framework/Versions/A/AppleJPEG
dyld: loaded: /System/Library/Frameworks/MetalPerformanceShaders.framework/Frameworks/MPSCore.framework/Versions/A/MPSCore
dyld: loaded: /System/Library/Frameworks/MetalPerformanceShaders.framework/Frameworks/MPSImage.framework/Versions/A/MPSImage
dyld: loaded: /System/Library/Frameworks/MetalPerformanceShaders.framework/Frameworks/MPSNeuralNetwork.framework/Versions/A/MPSNeuralNetwork
dyld: loaded: /System/Library/Frameworks/MetalPerformanceShaders.framework/Frameworks/MPSMatrix.framework/Versions/A/MPSMatrix
dyld: loaded: /System/Library/PrivateFrameworks/MetalTools.framework/Versions/A/MetalTools
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGLU.dylib
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGFXShared.dylib
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGLImage.dylib
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libCVMSPluginSupport.dylib
dyld: loaded: /System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libCoreVMClient.dylib
dyld: loaded: /usr/lib/libcups.2.dylib
dyld: loaded: /System/Library/Frameworks/Kerberos.framework/Versions/A/Kerberos
dyld: loaded: /System/Library/Frameworks/GSS.framework/Versions/A/GSS
dyld: loaded: /usr/lib/libresolv.9.dylib
dyld: loaded: /usr/lib/libiconv.2.dylib
dyld: loaded: /System/Library/PrivateFrameworks/Heimdal.framework/Versions/A/Heimdal
dyld: loaded: /usr/lib/libheimdal-asn1.dylib
dyld: loaded: /System/Library/Frameworks/OpenDirectory.framework/Versions/A/OpenDirectory
dyld: loaded: /System/Library/PrivateFrameworks/CommonAuth.framework/Versions/A/CommonAuth
dyld: loaded: /System/Library/Frameworks/OpenDirectory.framework/Versions/A/Frameworks/CFOpenDirectory.framework/Versions/A/CFOpenDirectory
dyld: loaded: /System/Library/Frameworks/SecurityFoundation.framework/Versions/A/SecurityFoundation
dyld: loaded: /System/Library/PrivateFrameworks/APFS.framework/Versions/A/APFS
dyld: loaded: /usr/lib/libutil.dylib
dyld: loaded: /System/Library/Frameworks/CoreAudio.framework/Versions/A/CoreAudio
dyld: loaded: /System/Library/Frameworks/AudioToolbox.framework/Versions/A/AudioToolbox
dyld: loaded: /System/Library/PrivateFrameworks/AppleSauce.framework/Versions/A/AppleSauce
dyld: loaded: /System/Library/PrivateFrameworks/LinguisticData.framework/Versions/A/LinguisticData
dyld: loaded: /usr/lib/libmarisa.dylib
dyld: loaded: /System/Library/PrivateFrameworks/Lexicon.framework/Versions/A/Lexicon
dyld: loaded: /usr/lib/libChineseTokenizer.dylib
dyld: loaded: /usr/lib/libcmph.dylib
dyld: loaded: /System/Library/PrivateFrameworks/LanguageModeling.framework/Versions/A/LanguageModeling
dyld: loaded: /System/Library/Frameworks/CoreData.framework/Versions/A/CoreData
dyld: loaded: /System/Library/PrivateFrameworks/CoreEmoji.framework/Versions/A/CoreEmoji
dyld: loaded: /System/Library/Frameworks/ServiceManagement.framework/Versions/A/ServiceManagement
dyld: loaded: /System/Library/PrivateFrameworks/BackgroundTaskManagement.framework/Versions/A/BackgroundTaskManagement
dyld: loaded: /usr/lib/libxslt.1.dylib
HTTP/1.0 400 Bad Request
Content-Type: text/html; charset=UTF-8
Referrer-Policy: no-referrer
Content-Length: 1555
Date: Thu, 21 Jun 2018 17:28:22 GMT

<!DOCTYPE html>
<html lang=en>
  <meta charset=utf-8>
  <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">
  <title>Error 400 (Bad Request)!!1</title>
  <style>
    *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png) no-repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png

real	0m0.479s
user	0m0.210s
sys	0m0.032s
```

### CGO_ENABLED=0 GOOS=darwin go build -o full-static -a -ldflags '-extldflags "-static"' main.go

```
$ time DYLD_PRINT_LIBRARIES=YES ./full-static 
dyld: loaded: /Users/XXX/workspace/go/src/github.com/XXX/XXX/./full-static
2018/06/21 10:30:20 EOF

real	0m29.780s
user	0m1.837s
sys	0m2.249s
```

```
$ time DYLD_PRINT_LIBRARIES=YES /usr/local/bin/helm init --client-only
dyld: loaded: /usr/local/bin/helm
Creating /Users/mavenlink/.helm/repository/repositories.yaml 
Adding stable repo with URL: https://kubernetes-charts.storage.googleapis.com 
Error: Looks like "https://kubernetes-charts.storage.googleapis.com" is not a valid chart repository or cannot be reached: Get https://kubernetes-charts.storage.googleapis.com/index.yaml: net/http: TLS handshake timeout

real	0m10.102s
user	0m0.682s
sys	0m0.761s
```

**Comment by bacongobbler** [comment]
Oh, if you're compiling Helm from scratch then this can definitely happen with older versions of Go since they didn't check the root cert chain which was fixed in later versions of Go.

What version of Go are you compiling Helm with? The official 2.9.1 release at https://github.com/kubernetes/helm/releases/tag/v2.9.1 was built with [Go 1.10](https://github.com/kubernetes/helm/blob/v2.9.1/.circleci/config.yml#L7) which we know fixes certain root cert chain check issues.

**Comment by diclophis** [comment]
@bacongobbler yea I am using the version of helm available for download from the github releases page:

```
$ shasum -a 256 /usr/local/bin/helm
bee3146de8b0cf940e74c9b1e7d3e23a1ffd8f7c4b8aad59c3e5a5152d7b818b  /usr/local/bin/helm
```

Also... I am using:

```
$ go version
go version go1.10.2 darwin/amd64
```

**Comment by bacongobbler** [comment]
Hmm that's the only time I've seen this crop up. Sorry :(

**Comment by diclophis** [comment]
@bacongobbler I am 99% certain this was some funky confluence of the mentioned go 1.10 tls issues mixed with a combination of locally installed duplicated developer site root certificates I had installed on my workstation... thanks for your quick responses... my engineering org is having great success with `helm`

**Comment by thedarkfalcon** [comment]
I'm having this same issue now :-/
I can openssl to the URL fine, but whatever I do Helm won't connect.

helm version
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}

helm init
Creating ~/test/repository/repositories.yaml
Adding stable repo with URL: https://kubernetes-charts.storage.googleapis.com
Error: Looks like "https://kubernetes-charts.storage.googleapis.com" is not a valid chart repository or cannot be reached: Get https://kubernetes-charts.storage.googleapis.com/index.yaml: dial tcp: i/o timeout

**Comment by diclophis** [comment]
@thedarkfalcon double check and confirm you dont have an abundance of "broken" or  "self-signed" ssl certs in your `Keychain.app`... I think the issue crops up when you have more than ~150 self-signed certs locally. I had this issue and I cleaned up a ton of development certs I had created and the issue went away.

**Comment by thedarkfalcon** [comment]
Oh sorry I forgot to mention, I'm not actually on OSX, I'm on ubuntu. This was the most relevant thread I could find though.

**Comment by diclophis** [comment]
oh interesting @thedarkfalcon ...

**Comment by ReSearchITEng** [comment]
I see the issue is closed, but I am also facing this.
helm 2.10.0 , on Centos7/RHEL7
If I try 2.9.1 and older helm versions, works perfect. There is something broken in 2.10.0, but I can't figure out what is going on .
(PS: proxy is being used, but this is the case for 2.9.1 as well, and there it works)

**Comment by bacongobbler** [comment]
@ReSearchITEng see https://github.com/helm/helm/issues/4326; there was a regression in v2.10.0 that has been fixed in master.

**Comment by bencurrerburgess** [comment]
I had the same issue using helm 2.10.0
Used the v2.11.0-rc.4 pre-release binary and it fixed the issue

**Comment by jisheng** [comment]
I'm running helm v2.11.0, and I still see the same error if I init with refresh or update repo.

wget works fine.

+++++++++++++++++++
Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
+++++++++++++++++++
helm repo update
Hang tight while we grab the latest from your chart repositories...
...Skip local chart repository
...Unable to get an update from the "stable" chart repository (https://kubernetes-charts.storage.googleapis.com):
	Get https://kubernetes-charts.storage.googleapis.com/index.yaml: proxyconnect tcp: EOF
Update Complete. \u2388 Happy Helming!\u2388

**Comment by bacongobbler** [comment]
that error looks different than what others have posted before.

>  proxyconnect tcp: EOF

Mind opening a new ticket for that? At the very least we can do some kind of root cause analysis to determine what triggers that error.

**Comment by jisheng** [comment]
filed #4732. thx

**Comment by bacongobbler** [comment]
@Krelion if you can point out where exactly in Helm /etc/resolv.conf gets rewritten, I'd be happy to take a look.

**Comment by sdake** [comment]
Note to others that run into this problem.  I have had this problem just this morning.  The root cause is that certs are not installed in ubuntu:bionic by default.

`apt-get install -y ca-certificates`

Within the docker build got me moving.

Cheers
-steve

**Comment by tibwiz** [comment]
I also faced the same issue with helm version 2.9.1 

Adding stable repo with URL: https://kubernetes-charts.storage.googleapis.com
Error: Looks like "https://kubernetes-charts.storage.googleapis.com" is not a valid chart repository or cannot be reached: Get https://kubernetes-charts.storage.googleapis.com/index.yaml: proxyconnect tcp: EOF

The solution mentioned in #4271 worked for me.

**Comment by tirelibirefe** [comment]
the problem is still exists.
Windows10 1909
version.BuildInfo{Version:"v3.0.1", GitCommit:"7c22ef9ce89e0ebeb7125ba2ebf7d421f3e82ffa", GitTreeState:"clean", GoVersion:"go1.13.4"}

**Comment by sedkis** [comment]
Exists for me as well, on Mac Os X

**Comment by maateen** [comment]
The following commands do the charm for me:

```shell
$ brew uninstall kubernetes-helm
$ brew install kubernetes-helm
```

**Comment by sdake** [comment]
There are several reasons this can happen

1. you have a proxy in your network. In this case, docker has instructions on using a proxy.
1. general network problems. these can be debugged with `curl -v https://address`. DNS may not be resolving, or if it is, that address may be `blackholed` - meaning blocked by your outgoing network.
1. and the more likely scenario, is if running ubuntu bionic, the ca-certs are not present by default in the image. I am not sure why the ca certs are not installed by default - they are even installed in the distroless container produced by google and necessary for anything hhtp(S) related to work.

For windows, I am not sure what the problem would be, other than that ca certs are not present on the host.

For mac, ca certs are installed by default.

Cheers
-steve

**Comment by maateen** [comment]
I am using MacOSX Catalina 10.15.3 and still facing the problems. So, I've replaced the stable repo with the Azure mirror.

Delete the original stable repo:

```bash
$ helm repo remove stable
```

Add Azure repo:

```bash
helm repo add stable http://mirror.azure.cn/kubernetes/charts/
```

**Comment by neil-n-brown** [comment]
I using the latest Helm Windows 10 installed by Choclately and I am encountering the same problem for all repos that I try to add 

helm repo add stable http://mirror.azure.cn/kubernetes/charts/

"Error: looks like "http://mirror.azure.cn/kubernetes/charts/" is not a valid chart repository or cannot be reached: Get http://mirror.azure.cn/kubernetes/charts/index.yaml: dial tcp XXX.XXX.XXX.XX:80: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond."

Same error for:  helm repo add stable https://kubernetes-charts.storage.googleapis.com

I have set proxy but same result.

Does anyone have any thoughts on this from a Windows perspective?

**Comment by maateen** [comment]
@neil-n-brown I am surprised too. A few days earlier, I got the same error from Azure. Whatever, I have updated the mirror repository in [my previous comment](https://github.com/helm/helm/issues/4251#issuecomment-611584176). Please check.

**Comment by neil-n-brown** [comment]
> @neil-n-brown I am surprised too. A few days earlier, I got the same error from Azure. Whatever, I have updated the mirror repository in [my previous comment](https://github.com/helm/helm/issues/4251#issuecomment-611584176). Please check.

Thanks for the reply @maateen much appreciated. Unfortunately I get the error:

Error: looks like "https://charts.ost.ai" is not a valid chart repository or cannot be reached: Get https://charts.ost.ai/index.yaml: dial tcp XXX.XXX.XXX.XXX:443: i/o timeout

But I can curl https://charts.ost.ai/index.yaml without issue.

**Comment by maateen** [comment]
@neil-n-brown  Please make sure you are using the latest helm. By the way, what's the output if you try to use these URLs over the VPN? (Use your favorite VPN)

**Comment by neil-n-brown** [comment]
> @neil-n-brown Please make sure you are using the latest helm. By the way, what's the output if you try to use these URLs over the VPN? (Use your favorite VPN)

Using the lasted helm 3.2.1. Can't test using a VPN unfortunately im on a company network. But I do not anticipate this to be a proxy issue as I tried to set the http and https proxy via command line. And I can curl or access the repos via the web browser without issue.

Do you know where helm (in windows 10) looks for its network configuration?

**Comment by maateen** [comment]
When it is Windows, I am out of clue. Sorry man!

**Comment by huide9** [comment]
The following link really helped me to solve this problem, wish it can help the coming guys: 

@VonC gave full explanation and @phiphu has the direct solution in the following link:
https://stackoverflow.com/questions/21181231/server-certificate-verification-failed-cafile-etc-ssl-certs-ca-certificates-c

**Comment by piyushpyoaknorth** [comment]
Hi,

I have tried the following things and still facing the issue on Mac.

Can you help?

1) Deleted old help repo and added new repo
```
        * helm repo remove stable
        * helm repo add stable https://charts.ost.ai
```
2) Solution mentioned in 
    * https://superuser.com/questions/1325221/linux-how-to-recursively-chmod-a-folder

```
        * echo $(echo -n | openssl s_client -showcerts -connect kubernetes-charts.storage.googleapis.com:443 2>/dev/null  | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p') >> /etc/ssl/certs/ca-certificates.crt
```

I also verified that the command 
``` openssl s_client -connect kubernetes-charts.storage.googleapis.com:443```

works as expected

```
CONNECTED(00000005)
depth=2 OU = GlobalSign Root CA - R2, O = GlobalSign, CN = GlobalSign
verify return:1
depth=1 C = US, O = Google Trust Services, CN = GTS CA 1O1
verify return:1
depth=0 C = US, ST = California, L = Mountain View, O = Google LLC, CN = *.storage.googleapis.com
verify return:1
---
Certificate chain
 0 s:/C=US/ST=California/L=Mountain View/O=Google LLC/CN=*.storage.googleapis.com
   i:/C=US/O=Google Trust Services/CN=GTS CA 1O1
 1 s:/C=US/O=Google Trust Services/CN=GTS CA 1O1
   i:/OU=GlobalSign Root CA - R2/O=GlobalSign/CN=GlobalSign
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIHCzCCBfOgAwIBAgIQdfo76CJBc+4CAAAAAHpJ4zANBgkqhkiG9w0BAQsFADBC
MQswCQYDVQQGEwJVUzEeMBwGA1UEChMVR29vZ2xlIFRydXN0IFNlcnZpY2VzMRMw
EQYDVQQDEwpHVFMgQ0EgMU8xMB4XDTIwMDkwMzA2MjcwOVoXDTIwMTEyNjA2Mjcw
OVowcjELMAkGA1UEBhMCVVMxEzARBgNVBAgTCkNhbGlmb3JuaWExFjAUBgNVBAcT
DU1vdW50YWluIFZpZXcxEzARBgNVBAoTCkdvb2dsZSBMTEMxITAfBgNVBAMMGCou
c3RvcmFnZS5nb29nbGVhcGlzLmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCC
AQoCggEBAMN9Ft7R13A/F4H1ncoNuSV+UJY7uUxHzkwq6EVP0qGyfi8k8F5Pb2vk
b62t5PbMKYVChGZFlFCPE0gQtZ8WlAAQyOEs9BBty2xUKxS6CVIFXpTm0eDPRuaP
C+sbBAW3W1GTRsv1sYDlYeKOvuxZTjN6wiPY2Q5tCniJixkNX/M0/LAhJ6rK/ltQ
yf/ahkcDCg9bGw+fgjqSEkGJHqlI+l0xQxAiZ0exkHp9sTFXyAblgUaXGCa8QnQk
8S/H0Hw859dB/0y239R7VnNFn8aB18EDo9q63LMbvvKLBIaf8cuhbSZkufLMkXDx
k/sZ9IiPJE6/+3zRRyI6afLtpmO52UMCAwEAAaOCA8swggPHMA4GA1UdDwEB/wQE
AwIFoDATBgNVHSUEDDAKBggrBgEFBQcDATAMBgNVHRMBAf8EAjAAMB0GA1UdDgQW
BBQ2KtUoRuckInz1+8KGDRUF4bjIDDAfBgNVHSMEGDAWgBSY0fhuEOvPm+xgnxiQ
G6DrfQn9KzBoBggrBgEFBQcBAQRcMFowKwYIKwYBBQUHMAGGH2h0dHA6Ly9vY3Nw
LnBraS5nb29nL2d0czFvMWNvcmUwKwYIKwYBBQUHMAKGH2h0dHA6Ly9wa2kuZ29v
Zy9nc3IyL0dUUzFPMS5jcnQwggGIBgNVHREEggF/MIIBe4IYKi5zdG9yYWdlLmdv
b2dsZWFwaXMuY29tgiQqLmFwcHNwb3QuY29tLnN0b3JhZ2UuZ29vZ2xlYXBpcy5j
b22CIiouY29tbW9uZGF0YXN0b3JhZ2UuZ29vZ2xlYXBpcy5jb22CKSouY29udGVu
dC1zdG9yYWdlLWRvd25sb2FkLmdvb2dsZWFwaXMuY29tgicqLmNvbnRlbnQtc3Rv
cmFnZS11cGxvYWQuZ29vZ2xlYXBpcy5jb22CICouY29udGVudC1zdG9yYWdlLmdv
b2dsZWFwaXMuY29tghAqLmdvb2dsZWFwaXMuY29tgiEqLnN0b3JhZ2UtZG93bmxv
YWQuZ29vZ2xlYXBpcy5jb22CHyouc3RvcmFnZS11cGxvYWQuZ29vZ2xlYXBpcy5j
b22CIGNvbW1vbmRhdGFzdG9yYWdlLmdvb2dsZWFwaXMuY29tghZzdG9yYWdlLmdv
b2dsZWFwaXMuY29tgg91bmZpbHRlcmVkLm5ld3MwIQYDVR0gBBowGDAIBgZngQwB
AgIwDAYKKwYBBAHWeQIFAzAzBgNVHR8ELDAqMCigJqAkhiJodHRwOi8vY3JsLnBr
aS5nb29nL0dUUzFPMWNvcmUuY3JsMIIBAgYKKwYBBAHWeQIEAgSB8wSB8ADuAHUA
B7dcG+V9aP/xsMYdIxXHuuZXfFeUt2ruvGE6GmnTohwAAAF0Uty2xAAABAMARjBE
AiBpVV+nx/Gha1aH7/LazbtLIuuEADTCGIT+p8j6BBkPOAIgM2s9urnxlEThivBM
cHn/0W8g6Lq/S13idoBXaHEok+wAdQBep3P531bA57U2SH3QSeAyepGaDIShEhKE
GHWWgXFFWAAAAXRS3La1AAAEAwBGMEQCID3Kd+gR+w3KZo55Ai5QJYKok50tZ8Y8
X6qsk6hUcUKqAiAMyl7Jwg/JiNV7dFCgPr9AqWt1fKWwa9BlgJZm0vgqhjANBgkq
hkiG9w0BAQsFAAOCAQEAxwfrG9QocOAth7aeDTYTOu5vAeaYy++j9oQm44LAUFMS
axjc8dxMHwQn+f7Oxeu9Tr5ggn6jQ10V0+V2HrMNxtqJFkS24AchoOPgRCsQHnMG
UUp7udu3uy5bFf+WD3MgiWoGn7quuD2zKhpmfmLMRozPQ6ylCg0HSV5areA5Vhj5
/ReK/cN8RzRMYJcMrBEVPd7MZ1uI8Oea4daYsWS0FS1iDoksj/6g+Z4sLRb+RJVV
UPwdaq6De4jv1g9qSFjdS8Ksky8ENaX/w73nJXBq1G220gGPf8u1zdRI+0j4FiVD
WrV7Ig+wnjChNm2gbuO39IupN6ZPZRIRPJDnr8icyw==
-----END CERTIFICATE-----
subject=/C=US/ST=California/L=Mountain View/O=Google LLC/CN=*.storage.googleapis.com
issuer=/C=US/O=Google Trust Services/CN=GTS CA 1O1
---
No client certificate CA names sent
Server Temp Key: ECDH, X25519, 253 bits
---
SSL handshake has read 3584 bytes and written 281 bytes
---
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-CHACHA20-POLY1305
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-CHACHA20-POLY1305
    Session-ID: BC5AABC8462E3535D22C1575C113FE02226128D3D1890AB32C7043C791DADB0B
    Session-ID-ctx: 
    Master-Key: 0E259E703B0CD1CE7B21D348B3895E8F567F45CD688D8AF79CAA65F101937FAB4E02CDD3384EC9445C06546A0B1D75CB
    TLS session ticket lifetime hint: 100800 (seconds)
    TLS session ticket:
    0000 - 01 25 df 63 11 70 49 c4-e8 7c 1a b0 fb f3 25 a4   .%.c.pI..|....%.
    0010 - c0 2e ff e8 19 ce 1e 36-35 a2 b0 91 e4 e4 b4 05   .......65.......
    0020 - ed f3 81 7f 7b e0 79 b3-f3 cd 65 90 b1 08 cb 3e   ....{.y...e....>
    0030 - cc d1 fa c1 63 63 f1 15-57 c4 e1 00 25 c5 d7 8d   ....cc..W...%...
    0040 - c8 c0 9c 61 91 47 24 89-ac 96 9d bc 56 b9 6b cc   ...a.G$.....V.k.
    0050 - 54 bb 5c 2a 3c 7d 62 9a-f4 85 c7 d7 35 ce b7 79   T.\*<}b.....5..y
    0060 - f3 33 72 f3 bf 01 e0 45-d9 59 dd a5 da 39 0f ec   .3r....E.Y...9..
    0070 - f9 5d 13 41 c3 00 cf 14-c0 13 83 bf 3b 4d a0 00   .].A........;M..
    0080 - 9a 52 41 29 40 d2 4d 7f-0a 72 f6 49 72 38 75 32   .RA)@.M..r.Ir8u2
    0090 - 3d f5 f9 20 1c 5e 04 46-ec 91 29 ae 8a ec 8c b7   =.. .^.F..).....
    00a0 - 73 3f 22 32 fb 53 e3 69-14 a6 89 e5 2c 68 75 0a   s?"2.S.i....,hu.
    00b0 - cc 8d 7d 3f e1 4a aa 9d-95 44 02 f5 69 4b 27 ca   ..}?.J...D..iK'.
    00c0 - ea f3 e3 c3 cf 60 27 6f-88 cc 43 b2 2d e0 20 83   .....`'o..C.-. .
    00d0 - 84 a6 6d 39 71 e2 84 94-7e 3d d9 aa ec            ..m9q...~=...

    Start Time: 1601537737
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
```
